src/deadline/client/api/_session.py

```diff
@@ -67,36 +60,32 @@ def get_boto3_session(
     if profile_name in ("(default)", "default", ""):
         profile_name = None
 
-    # If a config was provided, don't use the Session caching mechanism.
-    if config:
-        return boto3.Session(profile_name=profile_name)
-
     if force_refresh:
         invalidate_boto3_session_cache()
 
-    # If this is the first call or the profile name has changed, make a fresh Session
-    if not __cached_boto3_session or __cached_boto3_session_profile_name != profile_name:
-        __cached_boto3_session = boto3.Session(profile_name=profile_name)
-        __cached_boto3_session_profile_name = profile_name
+    return _get_boto3_session_for_profile(profile_name)
 
-    return __cached_boto3_session
 
+@lru_cache
+def _get_boto3_session_for_profile(profile_name: str):
+    session = boto3.Session(profile_name=profile_name)
 
-def invalidate_boto3_session_cache() -> None:
-    """
-    Invalidates the cached boto3 session and boto3 queue session.
-    """
-    global __cached_boto3_session
-    global __cached_boto3_session_profile_name
-    global __cached_boto3_queue_session
-    global __cached_farm_id_for_queue_session
-    global __cached_queue_id_for_queue_session
+    # By default, DCM returns creds that expire after 15 minutes, and boto3's RefreshableCredentials
+    # class refreshes creds that are within 15 minutes of expiring, so credentials would never be reused.
+    # Also DCM credentials currently take several seconds to refresh. Lower the refresh timeouts so creds
+    # are reused between API calls to save time.
+    # See https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L342-L362
+    credentials = session.get_credentials()
+    if isinstance(credentials, RefreshableCredentials):
```

Comment:
It is surprising to me that boto considers 10-15 minutes 'stale' and I wonder what the reason is. is there a risk to customers that are using other credential provider mechanisms? Should we further narrow down these credentials to just DCM-powered credentials?

Reply:
Looks like it was changed from 5 minutes to 15 minutes in this PR but there's not a justification for it that I can see: https://github.com/boto/botocore/pull/286

PRs since then have improved the refreshing mechanism but carried over the 15 minute timeout without commenting on it.

Reply:
I scoped this branch down to credentials whose method is "custom-process". It could include other types of process-provided creds, but I think the risk of them breaking is very small and the number of customers using other types of process-provided creds with the CLI is probably few. I looked for ways of detecting just DCM creds, but they'd require more complexity and refactoring which seemed like a bad tradeoff.